gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java

```diff
@@ -254,25 +256,27 @@ public static Properties getJobPropertiesFromJobDetail(JobDetail jobDetail) {
   @VisibleForTesting
   public static JobDataMap updatePropsInJobDataMap(JobDataMap jobDataMap,
       LeaseAttemptStatus.LeasedToAnotherStatus leasedToAnotherStatus, int schedulerMaxBackoffMillis) {
-    Properties prevJobProps = (Properties) jobDataMap.get(GobblinServiceJobScheduler.PROPERTIES_KEY);
+    JobDataMap newJobDataMap = (JobDataMap) jobDataMap.clone();
```

Comment:
In case you missed,
`Returns a **shallow** copy of this HashMap instance: **the keys and values themselves are not cloned**.`
so we should not use word "deep" in the javadoc

Reply:
btw, the current code should achieve the desired result. i am just commenting about choosing the right words.
i think deep copy was not needed here.